core/src/main/java/org/apache/iceberg/catalog/BaseSessionCatalog.java

```diff
@@ -62,6 +62,11 @@ public <T> T withContext(SessionContext context, Function<Catalog, T> task) {
     return task.apply(asCatalog(context));
   }
 
+  public void commitTransaction(SessionContext context, List<TableCommit> commits) {
```

Comment:
Why add this here? I would expect to either add this to the `SessionCatalog` interface and to `Catalog`, or to create optional interfaces to add this behavior.

Reply:
generally speaking, the plan was to add this to `SessionCatalog` and `Catalog` (which are both in the `api` module), but given that we currently have `TableMetadata` (which is in the `core` module) in `TableCommit`, this turned out to be difficult.

I've changed this up now so that it's only defined in `RESTSessionCatalog` and `RESTCatalog`. I think putting this method there is probably best for now and later we can move them to interfaces.